File: interviewai-frontend/types/interview.ts
export interface ResumeData {
  name: string;
  email: string;
  skills: string[];
  technologies: string[];
  projects: Array<{ name: string; description: string; tech_stack: string[] }>;
  experience: Array<{ company: string; role: string; duration: string }>;
  summary: string;
}

export interface ResumeUploadResponse {
  resume_id: string;
  resume_data: ResumeData;
  interview_topics: string[];
  message: string;
}

export type RoleType =
  | "sde_intern"
  | "frontend_developer"
  | "backend_developer"
  | "fullstack_developer"
  | "data_analyst";

export interface InterviewStartResponse {
  interview_id: string;
  first_question: string;
  topic: string;
  difficulty: string;
  total_planned_questions: number;
}

export interface EvaluationResult {
  technical_score: number;
  communication_score: number;
  confidence_score: number;
  answer_quality: "Excellent" | "Good" | "Fair" | "Poor";
  missing_concepts: string[];
  follow_up_required: boolean;
  difficulty_change: "increase" | "maintain" | "decrease";
  topic: string;
  next_question: string;
  feedback: string;
}

export interface AnswerResponse {
  evaluation: EvaluationResult;
  interview_complete: boolean;
  next_question: string | null;
  question_index: number;
}

export interface InterviewQuestion {
  question: string;
  topic: string;
  answer?: string;
  evaluation?: EvaluationResult;
  expectedConcepts?: string[];
}

export interface FinalReport {
  interview_id: string;
  candidate_name: string;
  role: string;
  date: string;
  overall_score: number;
  technical_score: number;
  communication_score: number;
  confidence_score: number;
  strengths: string[];
  weaknesses: string[];
  learning_roadmap: string[];
  hiring_recommendation: string;
  summary: string;
  pdf_url?: string;
}

export type InterviewStatus = "idle" | "uploading" | "planning" | "active" | "evaluating" | "completed";

File: interviewai-backend/app/schemas/interview.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from enum import Enum

class RoleType(str, Enum):
    SDE_INTERN = "sde_intern"
    FRONTEND = "frontend_developer"
    BACKEND = "backend_developer"
    FULLSTACK = "fullstack_developer"
    DATA_ANALYST = "data_analyst"

class DifficultyLevel(str, Enum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"

class InterviewStartRequest(BaseModel):
    resume_id: str
    role: RoleType
    candidate_name: str = ""

class AnswerSubmitRequest(BaseModel):
    interview_id: str
    question_index: int
    answer_text: str
    time_taken_seconds: Optional[int] = None

class EvaluationResult(BaseModel):
    technical_score: int  # 1-10
    communication_score: int  # 1-10
    confidence_score: int  # 1-10
    answer_quality: str  # "Excellent" | "Good" | "Fair" | "Poor"
    missing_concepts: List[str]
    follow_up_required: bool
    difficulty_change: str  # "increase" | "maintain" | "decrease"
    topic: str
    next_question: str
    feedback: str

class InterviewState(BaseModel):
    interview_id: str
    resume_id: str
    candidate_name: str
    role: str
    status: str  # "active" | "paused" | "completed"
    current_question_index: int
    questions: List[Dict[str, Any]]
    evaluations: List[Dict[str, Any]]
    interview_context: Dict[str, Any]
    created_at: str
    updated_at: str

class InterviewStartResponse(BaseModel):
    interview_id: str
    first_question: str
    topic: str
    difficulty: str
    total_planned_questions: int

class AnswerResponse(BaseModel):
    evaluation: EvaluationResult
    interview_complete: bool
    next_question: Optional[str]
    question_index: int
